### ISSUE TITLE
[Bug] Key override doesn't suppress Shift before firing target key

### ISSUE THREAD

**Issue by clbn** [qmk/qmk_firmware#16834]
## Describe the Bug

I'm trying to implement a simple key override with Shift mod, but Shift is not suppressed before firing the target key.

Moreover, while I can reproduce this behavior on one of my keyboards (Dactyl Manuform), I can't reproduce it on the other (Planck), where the KO works as intended.

### Minimal reproducible example

Let's say I want to override Shift + Comma, to output Square Bracket (`[`). This is the code I've added to `keymap.c`:

```
const key_override_t comma_bracket_override = ko_make_basic(MOD_MASK_SHIFT, KC_COMMA, KC_LBRC);

const key_override_t **key_overrides = (const key_override_t *[]){
    &comma_bracket_override,
    NULL
};
```

In my Planck, indeed, it outputs the "lowercase" _square_ bracket. In order to do this, it properly suppresses Shift (otherwise it would be an "uppercase" curly bracket instead):

```
Correct behaviour (Planck):

KEY-DOWN - QMK: KC_LSFT Event key: Shift       Code: ShiftLeft     KeyCode: 16
KEY-UP   - QMK: KC_LSFT Event key: Shift       Code: ShiftLeft     KeyCode: 16 in 151.000ms
KEY-DOWN - QMK: KC_LBRC Event key: [           Code: BracketLeft   KeyCode: 219
KEY-DOWN - QMK: KC_LSFT Event key: Shift       Code: ShiftLeft     KeyCode: 16
KEY-UP   - QMK: KC_LBRC Event key: {           Code: BracketLeft   KeyCode: 219 in 142.000ms
KEY-UP   - QMK: KC_LSFT Event key: Shift       Code: ShiftLeft     KeyCode: 16 in 56.000ms
```

However, in my Dactyl Manuform, it outputs an "uppercase" _curly_ bracket, because Shift is not suppressed before that:

```
Incorrect behaviour (DM):

KEY-DOWN - QMK: KC_LSFT Event key: Shift       Code: ShiftLeft     KeyCode: 16
KEY-DOWN - QMK: KC_LBRC Event key: {           Code: BracketLeft   KeyCode: 219
KEY-UP   - QMK: KC_LSFT Event key: Shift       Code: ShiftLeft     KeyCode: 16 in 310.000ms
KEY-DOWN - QMK: KC_LSFT Event key: Shift       Code: ShiftLeft     KeyCode: 16
KEY-UP   - QMK: KC_LBRC Event key: {           Code: BracketLeft   KeyCode: 219 in 140.000ms
KEY-UP   - QMK: KC_LSFT Event key: Shift       Code: ShiftLeft     KeyCode: 16 in 51.000ms
```

In this log, it seems like the rows 2 and 3 are swapped. Perhaps a race condition?

Not sure what other differences between these two keyboards are relevant. For reproducing this minimal example I created a new keymap (for `planck/rev6` and for `handwired/dactyl_manuform/5x6_5` respectively) and tried to add as little code as possible to the default (for Planck it was just adding the code above, for DM I had to convert keymap.json to keymap.c and then add the KO code). I'm guessing the fact that it's a split might play a role, so I tried to put all the relevant keys on the same half, and even disconnected the other half completely, but result is the same.

Please let me know if there's some QMK settings or anything else I should look into, in order to debug this.

## System Information

**Keyboard:** `handwired/dactyl_manuform/5x6_5`
**Operating system:** macOS
**`qmk doctor` output:** 
```
Ψ QMK Doctor is checking your environment.
Ψ CLI version: 1.0.0
Ψ QMK home: /Users/clbn/qmk_firmware
Ψ Detected macOS 12.3.
Ψ Git branch: master
Ψ Repo version: 0.16.9
⚠ Git has unstashed/uncommitted changes.
⚠ The official repository does not seem to be configured as git remote "upstream".
Ψ CLI installed in virtualenv.
Ψ All dependencies are installed.
Ψ Found arm-none-eabi-gcc version 10.3.1
Ψ Found avr-gcc version 9.3.0
⚠ We do not recommend avr-gcc newer than 8. Downgrading to 8.x is recommended.
Ψ Found avrdude version 6.4
Ψ Found dfu-util version 0.11
Ψ Found dfu-programmer version 0.7.2
Ψ Submodules are up to date.
Ψ QMK is ready to go, but minor problems were found
```

**Any keyboard related software installed?** 
 - [ ] AutoHotKey (Windows)
 - [x] Karabiner (macOS)
 - [ ] Other:

**Comment by clbn** [comment]
I found out it was actually **Karabiner-Elements** that caused this issue, somehow messing with the events order. Even though the only modification I had there was mapping Caps Lock to Backspace (meant for the other, QMK-incompatable, keyboard).

Unchecking DM keyboard in the Devices tab fixes the issue:

![image](https://user-images.githubusercontent.com/1071933/163356639-f9518fe0-8210-4533-bba0-fa1399c60a5f.png)

(Further testing showed that even with no modifications applied, Karabiner changes the key event order, just by "listening" to the keyboard, so unchecking it in the Devices list is the only thing that fixed it for me.)

Closing this now, but hoping it helps someone coming from Google with the same or similar issue.
